hudi-utilities/src/test/java/org/apache/hudi/utilities/TestHoodieSnapshotExporter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.utilities;
+
+import org.apache.hudi.utilities.HoodieSnapshotExporter.OutputFormatValidator;
+
+import com.beust.jcommander.ParameterException;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TestHoodieSnapshotExporter {
```

Comment:
unit test separated from its functional test class

Reply:
meaning this is not bringing up any new resources to run the test... 
I assume this the principle we will be following?

unit - testing basic functionality at the class level, potentially using mocks. Expected to finish quicker
functional - brings up the services needed and runs test without mocking
integration - runs subset of functional tests, on a full fledged enviroment with dockerized services 

Might be good to add such a doc somewhere.. may be in travis.yml or in README even.. so developers understand what test is what.

Reply:
@vinothchandar yes exactly. created https://issues.apache.org/jira/browse/HUDI-1034 for this